real_check/real_check.ps1

```diff
@@ -0,0 +1,57 @@
+#Copyright (c) Microsoft. All rights reserved.
+#Licensed under the MIT license. See LICENSE file in the project root for full license information.
+function Check-Symbols {
+    param(
+        [string] $lib
+    )   
+        try {
+            [string]$output = cmd.exe /c dumpbin /ALL $lib
+        }
+        catch {
+            Write-Error  "Unable to run dumpbin on given lib"
+            return 2
+        }
+        $start_token = "public symbols"
+        $end_token = "Archive member name"
+        try {
+            [int] $start = $output.IndexOf($start_token)
+            [int] $end = $output.IndexOf($end_token, $start)
+            $symbol_text = ($output.Substring($start + $start_token.Length+1, $end-$start-$start_token.Length-1))
+            $split_symbols = $symbol_text.Split('',[StringSplitOptions]::RemoveEmptyEntries)
+        }
+        catch {
+            Write-Error  "Unable to parse dumpbin output"
+            return 2
+        }
+        if($split_symbols.Length -lt 2) {
+            Write-Error  "Unable to parse dumpbin output"
+            return 2
+        }
+        
+        $real_prefix = "real_"
+        $symbol_table = New-Object -TypeName "System.Collections.Generic.HashSet[string]"
+        $duplicate_symbols = New-Object -TypeName "System.Collections.ArrayList"
+    
+        $return_code = 0
+        for($i=0; $i -lt $split_symbols.Length; $i++){
+            if($i % 2 -ne 0) {
```

Comment:
>if($i % 2 -ne 0) { [](http://example.com/codeflow?start=12&length=18)

Is this modulo 2 because of how the formatting of the output of dumpbin is?
In that case that part with adding a comment with an example would actually help.

Reply:
Added comment in next iteration.